File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnAudioWorkEdit.cs

namespace GracenoteSDK {

/**
*  @internal GnAudioWorkEdit @endinternal
*  represents  AudioWork's information returned by Gracenote.
*/
public class GnAudioWorkEdit : GnDataObject {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;

  internal GnAudioWorkEdit(global::System.IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnAudioWorkEdit_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnAudioWorkEdit obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnAudioWorkEdit() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnAudioWorkEdit(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public GnCreditEdit Credit(uint ord) {
    GnCreditEdit ret = new GnCreditEdit(gnsdk_csharp_marshalPINVOKE.GnAudioWorkEdit_Credit(swigCPtr, ord), true);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

  public void Origin(GnListElement originElement) {
    gnsdk_csharp_marshalPINVOKE.GnAudioWorkEdit_Origin(swigCPtr, GnListElement.getCPtr(originElement));
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public void Era(GnListElement eraElement) {
    gnsdk_csharp_marshalPINVOKE.GnAudioWorkEdit_Era(swigCPtr, GnListElement.getCPtr(eraElement));
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

/**
*  @internal CompositionForm @endinternal
*  Sets a list-based value by the list item Submit ID for COMPOSITION FORM. If the value does not
*   exist, it is added. If the value does exist, it is changed. If the role_submit_id is 0, the
*   value is deleted.
*  @param value set Value corresponding to the specified GnDataObject value key
*  <p><b>Remarks:</b></p>
*  Use this function to set a list-based Submit ID to COMPOSITION FORM, prior to adding the GnDataObject to a
*   parcel.
*/
  public string CompositionForm {
	/* csvarin typemap code */
	set 
	{
		System.IntPtr tempvalue = GnMarshalUTF8.NativeUtf8FromString(value);
		gnsdk_csharp_marshalPINVOKE.GnAudioWorkEdit_CompositionForm_set(swigCPtr, tempvalue);
		GnMarshalUTF8.ReleaseMarshaledUTF8String(tempvalue);
	}
 
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnAudioWorkEdit_CompositionForm_get(swigCPtr) );
	} 

  }

/**
*  @internal getTitleEdit @endinternal
*  Create an empty object of GnTitleEdit using TITLE OFFICIAL.
*  @return GnTitleEdit
*/
  public GnTitleEdit Title {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnAudioWorkEdit_Title_get(swigCPtr);
      GnTitleEdit ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnTitleEdit(cPtr, true);
      return ret;
    } 
  }

  public GnCreditEdit CreditAdd {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnAudioWorkEdit_CreditAdd_get(swigCPtr);
      GnCreditEdit ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnCreditEdit(cPtr, true);
      return ret;
    } 
  }

  public GnAudioWork GnAudioWork {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnAudioWorkEdit_GnAudioWork_get(swigCPtr);
      GnAudioWork ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnAudioWork(cPtr, true);
      return ret;
    } 
  }

}

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnAudioWorkEdit.java

package com.gracenote.gnsdk;

/** 
* 
*  represents  AudioWork's information returned by Gracenote. 
*/ 
 
public class GnAudioWorkEdit extends GnDataObject {
  private transient long swigCPtr;

  protected GnAudioWorkEdit(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnAudioWorkEdit_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnAudioWorkEdit obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnAudioWorkEdit(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public GnAudioWork gnAudioWork() {
    return new GnAudioWork(gnsdk_javaJNI.GnAudioWorkEdit_gnAudioWork(swigCPtr, this), true);
  }

/** 
* 
*  Create an empty object of {@link GnTitleEdit} using TITLE OFFICIAL. 
*  @return {@link GnTitleEdit} 
*/ 
 
  public GnTitleEdit title() throws com.gracenote.gnsdk.GnException {
    return new GnTitleEdit(gnsdk_javaJNI.GnAudioWorkEdit_title(swigCPtr, this), true);
  }

  public GnCreditEdit creditAdd() throws com.gracenote.gnsdk.GnException {
    return new GnCreditEdit(gnsdk_javaJNI.GnAudioWorkEdit_creditAdd(swigCPtr, this), true);
  }

  public GnCreditEdit credit(long ord) throws com.gracenote.gnsdk.GnException {
    return new GnCreditEdit(gnsdk_javaJNI.GnAudioWorkEdit_credit(swigCPtr, this, ord), true);
  }

  public void origin(GnListElement originElement) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnAudioWorkEdit_origin(swigCPtr, this, GnListElement.getCPtr(originElement), originElement);
  }

  public void era(GnListElement eraElement) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnAudioWorkEdit_era(swigCPtr, this, GnListElement.getCPtr(eraElement), eraElement);
  }

/** 
* 
*  Sets a list-based value by the list item Submit ID for COMPOSITION FORM. If the value does not 
*   exist, it is added. If the value does exist, it is changed. If the role_submit_id is 0, the 
*   value is deleted. 
*  @param value set Value corresponding to the specified {@link GnDataObject} value key 
*  <p><b>Remarks:</b></p> 
*  Use this function to set a list-based Submit ID to COMPOSITION FORM, prior to adding the {@link GnDataObject} to a 
*   parcel. 
*/ 
 
  public String compositionForm() {
    return gnsdk_javaJNI.GnAudioWorkEdit_compositionForm__SWIG_0(swigCPtr, this);
  }

  public void compositionForm(String value) throws com.gracenote.gnsdk.GnException {
    gnsdk_javaJNI.GnAudioWorkEdit_compositionForm__SWIG_1(swigCPtr, this, value);
  }

}
